src/main/java/com/ericsson/eiffel/remrem/semantics/SemanticsService.java

```diff
@@ -403,10 +409,7 @@ public String generateRoutingKey(JsonObject eiffelMessage, String tag, String do
      */
     public static Source setSerializerGav(Source source) {
         source = source == null ? new Source() : source;
-        Serializer serializer = source.getSerializer() == null ? new Serializer() : source.getSerializer();
-        serializer.setGroupId(semanticsGAV.getGroupId());
-        serializer.setArtifactId(semanticsGAV.getArtifactId());
-        serializer.setVersion(semanticsGAV.getVersion());
+        String serializer = source.getSerializer() == null ? new String() : source.getSerializer();
```

Comment:
set the serializer, i dont see we are setting serializer at all. serializer should be set in purl format.